File: content/courses/parallel-computing-introduction/codes/mpi_init_thread.cxx
#include <iostream>
#include <mpi.h>

using namespace std;

int main(int argc, char* argv[]) {
    int provided;
    MPI_Init_thread(&argc, &argv, MPI_THREAD_MULTIPLE, &provided);
    cout<<"Requested "<<MPI_THREAD_MULTIPLE<<" Provided "<<provided<<endl;
    MPI_Finalize();

    exit(0);
}

File: content/courses/parallel-computing-introduction/codes/mpi_init_thread.f90
program mpi_threads
use mpi_f08

integer ::  provided

   call MPI_INIT_THREAD(MPI_THREAD_MULTIPLE, provided)
   write(*,'(a,i2,a,i2)') "Requested ",MPI_THREAD_MULTIPLE," Provided ",provided
   call MPI_Finalize()

end program
